mtools/mloginfo/sections/cursor_section.py

```diff
@@ -0,0 +1,81 @@
+import re
+from collections import namedtuple
+from mtools.util import OrderedDict
+from mtools.util.grouping import Grouping
+from mtools.util.print_table import print_table
+from .base_section import BaseSection
+
+from collections import namedtuple
+from operator import itemgetter
+
+from .base_section import BaseSection
+from mtools.util import OrderedDict
+from mtools.util.grouping import Grouping
+from mtools.util.print_table import print_table
+
+LogTuple = namedtuple('LogTuple', ['datetime', 'cursorid', 'reapedtime'])
+
+
+def op_or_cmd(le):
+    return le.operation if le.operation != 'command' else le.command
+
+# SERVER-28604 - Logging Reaped cursor information when idle
+
+class CursorSection(BaseSection):
+    """CursorSection class."""
+
+    name = 'cursor'
+
+    def __init__(self, mloginfo):
+        BaseSection.__init__(self, mloginfo)
+        helptext = 'outputs statistics about cursor'
+        self.mloginfo.argparser_sectiongroup.add_argument('--cursor', action='store_true', help=helptext)
+
+    @property
+    def active(self):
+        """Return boolean if this section is active."""
+        return self.mloginfo.args['cursor']
+
+    def run(self):
+        """Run this section and print out information."""
+        grouping = Grouping(group_by=lambda x: (x.datetime, x.cursorid, x.reapedtime))
+        logfile = self.mloginfo.logfile
+
+        if logfile.start and logfile.end:
+            progress_start = self.mloginfo._datetime_to_epoch(logfile.start)
+            progress_total = (self.mloginfo._datetime_to_epoch(logfile.end) - progress_start)
+        else:
+            self.mloginfo.progress_bar_enabled = False
+
+        for i, le in enumerate(logfile):
+            if ('Cursor id' in le.line_str):
+                lt = LogTuple(le.datetime, le.cursor, le._reapedtime)
+                grouping.add(lt)
+
+        grouping.sort_by_size()
+
+        # clear progress bar again
+        if self.mloginfo.progress_bar_enabled:
+            self.mloginfo.update_progress(1.0)
+
+        # no cursor information in the log file
+        if not len(grouping):
+            print('no cursor information found.')
+            return
+
+        titles = ['datetime', 'cursorid', 'reapedtime']
+
+        table_rows = []
+        # using only important key-values
+        for g in grouping:
+            # calculate statistics for this group
+            datetime, cursorid, reapedtime = g
```

Comment:
⚠️  don't create variables with names of modules or builtins (datetime in this case), call it datetime_ for example